packages/nx/src/config/project-graph.ts

```diff
@@ -107,13 +107,16 @@ export interface ProjectGraphProjectNode<T = any> {
 
 /**
  * A node describing an external dependency
+ * `name` has as form of `npm:packageName` for primary dependencies or `npm:packageName@version` for hoisted ones
```

Comment:
Why can't we always use the @version version?

Reply:
We could, but in this case, we would need to add an additional hash table that links `packageName` to `packageName@version` since our project imports have no notion of versioning.

If `externalNodes` contain multiple versions of the same package, we need to be able to know and quickly access the root one.

Reply:
Currently our codebase uses `externalNodes[packageName]` in several places.

Changing this would affect too many projects.